### PR TITLE
Add NRI WebAssembly blog post

### PR DESCRIPTION
**Pull request by saschagrunert**
Adding a blog post about the NRI Wasm functionality. My goal is to publish it via the CNCF blog after general consensus: https://www.cncf.io/blog

/hold

### CODE REVIEW

**Comment by saschagrunert** [comment]
PTAL @mikebrow @askervin @klihub @haircommander

File: blog.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+## How it works
+
+The required change for the NRI landed with [Pull Request
+containerd/nri#121][pr121]. This change adds a [go-plugin][go-plugin] mechanism
+to the NRI. Each plugin gets compiled to Wasm, which means that it is
+size-efficient, memory-safe, automatically sandboxed and highly portable out of
+the box! The plugin system works in the same way as the NRI by using [Protocol
+Buffers][protobuf]. This means, that the NRI can reuse the existing API for
+gRPC, while the communication will happen in memory and not over the Remote
```
Comment (klihub) [inline blog.md:46]: ttRPC used for RPC.

**Comment by klihub** [comment]
@saschagrunert Thanks you ! Looks really good IMO. I only have a few gRPC vs. ttRPC nits, same as in #125.

File: blog.md (hunk 1 above)
Comment (saschagrunert) [reply blog.md]: Thanks, fixed as suggested.

**Review by mikebrow — APPROVED** [review]
LGTM nice.. 

love the part about pulling plugins as artifacts.. We've been talking about doing that for cni/cdi as well.

**Comment by klihub** [comment]
@saschagrunert I converted this to prevent it from accidentally getting merged, since my impression is that your intention is to get the content reviewed as a PR but then publish it as a CNCF blog post without merging it to the repo. If that's not the case, just flip it back to a ready-for-review state...

**Comment by saschagrunert** [comment]
Let’s close, I’m going to reach out to the CNCF today wrt a publication date.

File: blog.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+```bash
+make $(pwd)/build/bin/wasm TINYGO_DOCKER=1
+```
+
+In the future it may be possible to cross compile plugins using `GOOS=wasip1
+GOARCH=wasm go build`, but that is not implemented yet (see
+[knqyf263/go-plugin#58][go-plugin-58]).
+
+[example]: https://github.com/containerd/nri/blob/dd57194/plugins/wasm/plugin.go
```
Comment (askervin) [inline blog.md:82]: Github gives a warning about this link:

This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.

...it'll be probably safer to link to a commit in the main branch, like you have in the other links below.

**Review by askervin — COMMENTED** [review]
Very nice blog post, @saschagrunert! I hope it'll get a lot of attention.

Added one comment regarding the design. Feel free to decide if and how you'll address it. And found one suspicious link to github.

Looks great to me. :)

File: blog.md (hunk 2 above)
Comment (saschagrunert) [reply blog.md:82]: I'll update the link, thank you!

File: blog.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+sandbox _creation_, _stopping_ or _removal_, while corresponding data are the
+_name_, _namespace_ or corresponding _annotations_.
+
+[api]: https://github.com/containerd/nri/blob/eaf78a9/pkg/api/api.proto
+[ttrpc]: https://github.com/containerd/ttrpc
+[grpc]: https://grpc.io
+
+On one hand plugins written as daemons have the benefit of persisting the
+current state out of the box, while on the other hand they come with a
+performance and management overhead. For that reason, the NRI also supports
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply blog.md:29]: Thank you for the clarification, I'll update the section.

**Pull request closed without merging**